driver-core/src/main/java/com/datastax/driver/core/ThreadingOptions.java

```diff
@@ -0,0 +1,143 @@
+    // Kept for backward compatibility, but this should be customized via this class now
+    private static final int NON_BLOCKING_EXECUTOR_SIZE = SystemProperties.getInt(
+            "com.datastax.driver.NON_BLOCKING_EXECUTOR_SIZE", Runtime.getRuntime().availableProcessors());
+    private static final int DEFAULT_THREAD_KEEP_ALIVE_SECONDS = 30;
+
+    /**
+     * Builds a thread factory for the threads created by a given executor.
+     * <p/>
+     * This is used by the default implementations in this class, and also internally to create the Netty I/O pool.
+     *
+     * @param clusterName  the name of the cluster, as specified by
+     *                     {@link com.datastax.driver.core.Cluster.Builder#withClusterName(String)}.
+     * @param executorName a name that identifies the executor.
+     * @return the thread factory.
+     */
+    public ThreadFactory threadFactory(String clusterName, String executorName) {
+        return new ThreadFactoryBuilder()
+                .setNameFormat(clusterName + "-" + executorName + "-%d")
+                // Back with Netty's thread factory in order to create FastThreadLocalThread instances. This allows
+                // an optimization around ThreadLocals (we could use DefaultThreadFactory directly but it creates
+                // slightly different thread names, so keep we keep a ThreadFactoryBuilder wrapper for backward
+                // compatibility).
+                .setThreadFactory(new DefaultThreadFactory("ignored name"))
+                .build();
+    }
+
+    /**
+     * Builds the generic administration executor, used for tasks such as triggering registered
+     * {@link SchemaChangeListener}s, reacting to node state changes, and metadata updates.
+     * <p/>
+     * The default implementation sets the pool size to the number of available cores.
+     *
+     * @param clusterName the name of the cluster, as specified by
+     *                    {@link com.datastax.driver.core.Cluster.Builder#withClusterName(String)}.
+     * @return the executor.
+     */
+    public ThreadPoolExecutor executor(String clusterName) {
+        ThreadPoolExecutor executor = new ThreadPoolExecutor(
+                NON_BLOCKING_EXECUTOR_SIZE, NON_BLOCKING_EXECUTOR_SIZE,
+                DEFAULT_THREAD_KEEP_ALIVE_SECONDS, TimeUnit.SECONDS,
+                new LinkedBlockingQueue<Runnable>(),
+                threadFactory(clusterName, "worker"));
+        executor.allowCoreThreadTimeOut(true);
+        return executor;
+    }
+
+    /**
+     * Builds the executor used to block on new connections before they are added to a pool.
+     * <p/>
+     * The default implementation uses 2 threads.
+     *
+     * @param clusterName the name of the cluster, as specified by
+     *                    {@link com.datastax.driver.core.Cluster.Builder#withClusterName(String)}.
+     * @return the executor.
+     */
+    public ThreadPoolExecutor blockingExecutor(String clusterName) {
```

Comment:
I like that the names are 1:1 with the names in `Cluster.manager`.  I expect the people who will use this will will need to look at the source code anyways so it makes sense to sync them up.  I think in a future major release where we refactor our thread pools I agree it would be good to make the names more explicit.